src/providers/TransactionBuilderContextProvider.tsx

```diff
@@ -644,7 +650,117 @@ const TransactionBuilderContextProvider = ({
       setIsSubmitting(false);
-    } else {
+
+    }
+    else if (crossChainActions[0].type == TRANSACTION_BLOCK_TYPE.PLR_DAO_STAKE) {
+      let crossChainAction = crossChainActions[0];
+
+      if (!crossChainAction.receiveAmount) {
+        showAlertModal('Failed to get receiveAmount');
+        setIsSubmitting(false);
+        return;
+      }
+
+      let result: {
+        transactionHash?: string;
+        errorMessage?: string;
+      };
+
+      result = crossChainAction.useWeb3Provider
+        ? await submitWeb3ProviderTransactions(
+          getSdkForChainId(crossChainAction.chainId) as Sdk,
+          web3Provider,
+          crossChainAction.transactions,
+          crossChainAction.chainId,
+          providerAddress,
+        )
+        : await submitEtherspotAndWaitForTransactionHash(
+          getSdkForChainId(crossChainAction.chainId) as Sdk,
+          crossChainAction.transactions,
+          crossChainAction.gasTokenAddress ?? undefined,
+        );
+      if (
+        result?.errorMessage ||
+        (!result?.transactionHash?.length)
+      ) {
+        // showAlertModal(result.errorMessage ?? 'Unable to send transaction!');
+        setIsSubmitting(false);
+        return;
+      }
+
+      let flag = 1, errorOnLiFi;
+      while (flag) {
+        try {
+          const status = await getCrossChainStatusByHash(getSdkForChainId(plrDaoAsset.chainId) as Sdk, crossChainAction.chainId, plrDaoAsset.chainId, result.transactionHash, crossChainAction.bridgeUsed)
+          if (status?.status == "DONE" && status.subStatus == "COMPLETED") {
+            flag = 0;
+          } else if (status?.status === "FAILED") {
+            errorOnLiFi = 'Transaction Failed on LiFi'
+            flag = 0
+          }
+          await sleep(30);
+        } catch (err) {
+          errorOnLiFi = 'Transaction Failed on LiFi'
+          flag = 0;
+        }
+      }
+
+      if (errorOnLiFi) {
+        showAlertModal(errorOnLiFi);
+        setIsSubmitting(false);
+        return;
+      }
+
+      const estimateGas = await estimateCrossChainAction(
+        getSdkForChainId(CHAIN_ID.POLYGON),
+        web3Provider,
+        crossChainAction.destinationCrossChainAction[0],
+        providerAddress,
+        accountAddress,
+      );
+
+      const stakingTxns = await plrDaoStaking(null, BigNumber.from(crossChainAction.receiveAmount).sub(utils.parseUnits('0.02', 6)).sub(estimateGas.feeAmount ?? '0').toString(), transactionBlocks[0].type === "PLR_DAO_STAKE" ? transactionBlocks[0].values?.receiverAddress : '', getSdkForChainId(plrDaoAsset.chainId))
+      if (stakingTxns.errorMessage) {
+        showAlertModal(stakingTxns.errorMessage);
+        setIsSubmitting(false);
+        return;
+      }
+
+      const estimated = await estimateCrossChainAction(
+        getSdkForChainId(plrDaoAsset.chainId),
+        web3Provider,
+        crossChainAction.destinationCrossChainAction[0],
+        providerAddress,
+        accountAddress,
+      );
```

Comment:
We should add error checking here, as any of these `await`'s can fail, or you can check the results further down the code